gqcp/include/Processing/Properties/DysonOrbital.hpp

```diff
@@ -96,65 +96,124 @@ class DysonOrbital {
                 sign *= -1;
                 size_t p = onv.occupationIndexOf(e);
                 onv.annihilate(p);
 
-                // Now, we calculate the overlap in the (N-1)-'target' ONV basis.
-                // In order to access the correct coefficients for the, we need the address of the resulting (annihilated) ONV inside the 'target' ONV basis.
-                size_t address = target_onv_basis2.addressOf(onv.unsignedRepresentation());
+                // Now, we calculate the overlap in the (N-1)-electron 'target' ONV basis.
+                // In order to access the correct coefficients for the, we need the address of the resulting (annihilated) ONV inside the 'target' (N-1)-electron ONV basis.
+                size_t address = target_onv_basis_I.addressOf(onv.unsignedRepresentation());
 
                 double coeff = 0;
-                for (size_t Ip = 0; Ip < passive_onv_basis1.dimension(); Ip++) {                                                             // Ip loops over the addresses of the passive ONV basis.
-                    coeff += sign * ci_coeffs1(It * target_mod + Ip * passive_mod1) * ci_coeffs2(address * target_mod + Ip * passive_mod2);  // Access the indices of the coefficient vectors.
+                for (size_t Jp = 0; Jp < passive_onv_basis_J.dimension(); Jp++) {                                                                // Jp loops over the addresses of the passive ONV basis.
+                    coeff += sign * ci_coeffs_J(Jt * target_mod + Jp * passive_mod_J) * ci_coeffs_I(address * target_mod + Jp * passive_mod_I);  // Access the indices of the coefficient vectors.
                 }
                 dyson_coeffs(p) += coeff;
                 onv.create(p);  // Allow the iteration to continue with the original ONV.
             }
 
-            if (It < target_onv_basis1.dimension() - 1) {  // Prevent the last permutation from occurring.
-                target_onv_basis1.transformONVToNextPermutation(onv);
+            if (Jt < target_onv_basis_J.dimension() - 1) {  // Prevent the last permutation from occurring.
+                target_onv_basis_J.transformONVToNextPermutation(onv);
             }
-        }  // Target address (It) loop.
+        }  // Target address (Jt) loop.
+
+        return DysonOrbital<Scalar>(dyson_coeffs);
+    }
+
+    /**
+     *  Create a Dyson orbital from the formula for its amplitudes `<N_1|a_p|N>`.
+     * 
+     *  @param linear_expansion_J        The N-electron wave function in a spin-unresolved ONV basis.
+     *  @param linear_expansion_I        The (N-1)-electron wave function in a spin-unresolved ONV basis. It should be expressed in the same orbital basis as the N-electron wave function.
+     *
+     *  @return A Dyson orbital incorporating Dyson amplitudes.
+     */
+    static DysonOrbital<Scalar> TransitionAmplitudes(const LinearExpansion<SpinUnresolvedONVBasis>& linear_expansion_J, const LinearExpansion<SpinUnresolvedONVBasis>& linear_expansion_I) {
```

Comment:
I see that you've added the API `TransitionAmplitudes` for given spin-unresolved arguments.

I'm wondering (since we're doing similar things in other parts of the code): do `DysonOrbital`s have fundamentally the same interface (i.e. which methods should you be able to call on them) for spin-resolved and the spin-unresolved cases? If they do, this approach is fine. If they don't, it's better to design a new class.

Reply:
Yes, I think that, other than their (named) constructor, the spin-resolved and -unresolved Dyson orbitals have the same dimension and same behaviour.

Reply:
However, we are using spin-orbitals as an underlying basis for for the `SpinResolvedONVBasis` meaning that we should treat an annihilation on an `alpha` electron differently than an annihilation on a `beta` electron. `TransitionAmplitudes (SpinResolved)` on the other hand does not make a distinction between spin compounds and if we want to describe Dyson **spin-orbitals**, we should separate `alpha` and `beta` contributions.

Reply:
It is not completely clear to me how a Dyson spin-orbital should be defined. [Ortiz2020a](https://doi.org/10.1063/5.0016472) uses a rather vague definition: "Let Dyson spin-orbitals be expanded in terms of `M` orthonormal functions..." (eq. 33). Since eq. (1) in this article has an integral over `x` (space-spin coordinate), it seems that we should perhaps make separate classes for `SpinResolved` and `SpinUnresolved` `DysonOrbital`s.

Reply:
As for the methods that call upon them, they should be quite similar, if not the same. I have not yet seen different approaches for spin-resolved and -unresolved cases in the literature, but maybe I am not familiar enough with the topic yet.

Reply:
To prevent this discussion from getting out of hand, see the specific issue on this topic https://github.com/GQCG-res/dariatols_knowdes/issues/15.